### DISCUSSION TITLE
Lag spike on beatmap start

### DISCUSSION THREAD

**Discussion by chayleaf** [ppy/osu#15416] [Q&A] [answered]
(Linux) I'm a taiko player, and I tried switching to lazer full-time. On some maps (or maybe on all maps, not sure) the game lags within the first ~10-20 seconds of a map, and I have to restart it multiple times to get it to lag somewhere where it won't cause a miss. I suspect this is due to some poor GC scheduling.

**Comment by bdach** [comment, marked as answer]
Please read: https://github.com/ppy/osu/issues/11800

In particular:

> there seems to be a difference in how the GC tunes its gen1 heap size on Linux vs Windows, resulting in making it too large on Linux and triggering collections less frequently than we'd like

You can try setting `COMPlus_GCGen0MaxBudget=600000` as an envvar. We may bake that into the appimage at some point if the situation does not improve on the .NET side.

**Comment by chayleaf** [comment]
I'll check it out. It's worth noting it doesn't happen late into the map, only at the start.
